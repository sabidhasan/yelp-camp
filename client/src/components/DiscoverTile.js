import React from 'react'
import LazyLoad from 'react-lazyload';
import { Link } from 'react-router-dom'
import sunsetImage from '../images/test2.jpg'

const DiscoverTile = (props) => {
  return (
    <div className='DiscoverTile flex-center'>
      <LazyLoad once>
        <img className='DiscoverTile__img' src={sunsetImage} alt="Sunset over mountains"/>
      </LazyLoad>
      <span className='DiscoverTile__text'>
        Find campgrounds in your province, and across Canada
      </span>
      <Link
        to='/discover'
        className='DiscoverTile__button btn btn--large'
      >
        DISCOVER
      </Link>
    </div>
  )
}

export default DiscoverTile
