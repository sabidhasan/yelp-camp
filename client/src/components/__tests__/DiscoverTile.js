import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DiscoverTile from '../DiscoverTile'

configure({ adapter: new Adapter() });

describe('<DiscoverTile />', () => {
  let component;

  beforeEach(() => {
    component = shallow(<DiscoverTile />)
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('DiscoverTile component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
