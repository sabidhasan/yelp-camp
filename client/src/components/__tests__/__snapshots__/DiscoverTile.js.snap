// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoverTile /> DiscoverTile component snapshot matches what is expected 1`] = `
<div
  className="DiscoverTile flex-center"
>
  <LazyLoad
    offset={0}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="Sunset over mountains"
      className="DiscoverTile__img"
      src="test2.jpg"
    />
  </LazyLoad>
  <span
    className="DiscoverTile__text"
  >
    Find campgrounds in your province, and across Canada
  </span>
  <Link
    className="DiscoverTile__button btn btn--large"
    replace={false}
    to="/discover"
  >
    DISCOVER
  </Link>
</div>
`;
